UnityProject/Assets/Scripts/Shuttles/ShuttleFuelSystem.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Atmospherics;
+using Mirror;
+
+
+/// <summary>
+/// Used to monitor the fuel level and to remove fuel from the canister and also stop the shuttle if the fuel has run out
+/// </summary>
+public class ShuttleFuelSystem : ManagedNetworkBehaviour
+{
+	public float FuelLevel;
+	public ShuttleFuelConnector Connector;
+	public MatrixMove MatrixMove;
+	public float FuelConsumption;
+	public float massConsumption = 2.5f;
+
+	public float CalculatedMassConsumption;
+
+	public float OptimumMassConsumption = 0.5f;
+
+
+	protected override void OnEnable() {
+		base.OnEnable();
+		if (MatrixMove == null) {
+			MatrixMove = this.GetComponent<MatrixMove>();
+			MatrixMove.shuttleFuelSystem = this;
+		}
+	}
+
+	public override void UpdateMe()
+	{
+
+		if (Connector.canister != null)
+		{
+			FuelConsumption =  MatrixMove.State.Speed / 25f;
+			if (MatrixMove.isMovingServer && MatrixMove.RequiresFuel)
+			{
+				FuelCalculations();
+			}
+			if (IsFuelled())
+			{
+				MatrixMove.IsFueled = true;
+			}
+			FuelLevel = Connector.canister.container.GasMix.GetMoles(Gas.Plasma) / 60000;
+		}
+		else
+		{
+			FuelLevel = 0f;
+			MatrixMove.IsFueled = false;
+			if (MatrixMove.isMovingServer && MatrixMove.RequiresFuel)
+			{
+				MatrixMove.StopMovement();
+			}
+		}
+		if (FuelLevel > 1)
+		{
+			FuelLevel = 1;
+		}
+		//Logger.Log("FuelLevel " + FuelLevel.ToString());
+	}
+
+	bool IsFuelledOptimum()
+	{
+		var Plasma = Connector.canister.container.GasMix.GetMoles(Gas.Plasma);
		var Oxygen = Connector.canister.container.GasMix.GetMoles(Gas.Oxygen);
+		var Ratio = ((Plasma / Oxygen) / (7f / 3f));
+		//Logger.Log("Ratio > " + Ratio);
+		Ratio = Ratio * 2;
+		//Logger.Log("Ratio1 > " + Ratio);
+		if (Ratio > 1)
+		{
+			Ratio = Ratio - 1;
+		}
+		else { 
+			Ratio = Ratio / 5;
+		}
+
+		if (Ratio > 1)
+		{
+			Ratio = 1 / Ratio;
+		}
+		//Logger.Log("Ratio2 > " + Ratio);
+
+		var CMassconsumption = 1 / Ratio;
+		if (CMassconsumption > 6)
+		{
+			CMassconsumption = 6;
+		}
+		//Logger.Log("Ratio3 > " + Ratio);
+		CalculatedMassConsumption = (CMassconsumption * OptimumMassConsumption * FuelConsumption);
+
+		if ((Plasma > (CalculatedMassConsumption)  * (7 / 10)) && (Oxygen > (CalculatedMassConsumption)  * (3 / 10)))

```

Comment:
use 0.7f and 0.3f in this file, diving 7 and 10 **integers** results in 0, see for yourself:
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUHjAZgAIBvPYy4jdMiqh1TATgApiB2AekwAZiAlAG56lAL54xQA=
again, proper IDEs warn you about that but it's actually basic theory